lib/mimemagic.rb

```diff
@@ -88,6 +111,23 @@ def self.by_magic(io)
       end
     mime && new(mime[0])
   end
+  
+  # Validate MIME by magic
+  def self.validate_by_magic(io, mime)
+    magic = MAGIC.find { |type, matches| type == mime.type }
+    return false if magic.nil?
+    matches = magic[1]
+    
+    unless io.respond_to?(:seek) && io.respond_to?(:read)
+      str = io.respond_to?(:read) ? io.read : io.to_s
+      str = str.force_encoding(Encoding::BINARY) if str.respond_to? :force_encoding
+      magic_match_str(str, matches)
+    else
+      io.binmode
+      io.set_encoding(Encoding::BINARY) if io.respond_to?(:set_encoding)
+      magic_match_io(io, matches)
+    end
+  end
 
```

Comment:
Is it possible that you try to reuse by_magic somehow or write a helper function which is used by by_magic and by by_path_magic? I think as you have it now, we have code duplication.